app/code/Magento/Checkout/view/frontend/web/js/view/form/element/email.js

```diff
@@ -192,7 +192,20 @@ define([
          * @returns {Boolean} - initial visibility state.
          */
         resolveInitialPasswordVisibility: function () {
-            if (checkoutData.getInputFieldEmailValue() !== '' && checkoutData.getCheckedEmailValue() === '') {
+            // If customer already logged in so we will hide password field
+            if (checkoutData.getInputFieldEmailValue() !== '' &&
+                customer.isLoggedIn()) {
+                return false;
+            }
+
+            // In this case user put email but not logged in yet password field should not show
+            if (checkoutData.getInputFieldEmailValue() !== '' &&
```

Comment:
Are you sure that we need to hide the password field when a user puts an email to the email field?

Reply:
this offered fix will not solve the issue and probably creates more issues . if customer is logged in theres never a e-mail field to fill anyway so the first condition is also bit odd 

if field is filled (like on page load from cache) it needs to be validated against registration status all over again . Customer could go back register on another window etc and reload the page and your check would fail. 

Also try to dive in to the other issue referenced that created the issue , it probably needs to cover the issue reported there as well.